packages/generator/src/entity/imports.ts

```diff
@@ -26,13 +27,17 @@ export function importDeclarations(
       [
         ...corePropertyTypeImportNames(entity.properties),
         ...corePropertyFieldTypeImportNames(entity.properties),
-        ...coreNavPropertyFieldTypeImportNames(entity.navigationProperties),
+        ...coreNavPropertyFieldTypeImportNames(
+          entity.navigationProperties,
+          service
```

Comment:
same here. If you just use the version in the end to decide which type of link to put, only pass the version.